python/hawkey/query-py.cpp

```diff
@@ -955,7 +956,7 @@ add_nevra_or_other_filter(_QueryObject *self, PyObject *args)
         if (!PyArg_ParseTuple(args, "s", &name))
             return NULL;
 
-        HyNevra out_nevra = hy_nevra_create();
+        HyNevra out_nevra = new Nevra;
```

Comment:
But if the Python bindings directly use C++, that seems to contradict the "libdnf will have a C API" part right?  Or do we see the Python bindings using C++ as a transient state until SWIG or something?

Reply:
@cgwalters We want to use SWIG for generating API for Python, Perl, .... We also assume supporting C API. 
But we can't do all in one day. Lindnf is under development now.

Reply:
I think @cgwalters's point here is that it makes sense to keep using only the C API in Python bindings, in order to ensure that the C API is fully functional. If DNF starts using C++ API (through Python bindings), then the C API is surely going to regress at some point, at it's less tested.

Reply:
I understand your concern. I hope that at the end will be one well tested C++ code. And C API will be thin (as thin as possible) wrapper which will be use the tested code.

Reply:
Right; I personally have never tried to maintain a C++ library with a C interface, but in just thinking about it briefly it seems like it would require rather strict discipline.  A lot of the advantages of C++ get murkier.

Anyways I'm not opposing this PR.